org.gnome.gnome-latex.json

```diff
@@ -0,0 +1,123 @@
+{
+  "app-id" : "org.gnome.gnome-latex",
+  "runtime" : "org.gnome.Sdk",
+  "runtime-version" : "3.30",
+  "platform-extensions" : [
+    "org.freedesktop.Sdk.Extension.texlive"
+  ],
+  "sdk" : "org.gnome.Sdk",
+  "command" : "gnome-latex",
+  "finish-args" : [
+    "--socket=x11",
+    "--socket=wayland",
+    "--filesystem=home",
+    "--share=ipc",
+    "--socket=session-bus",
+    "--filesystem=xdg-run/dconf",
+    "--talk-name=ca.desrt.dconf",
+    "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+    "--talk-name=org.freedesktop.Notifications",
+    "--env=PATH=/usr/bin:/app/bin:/usr/lib/sdk/texlive/bin/x86_64-linux"
+  ],
+  "cleanup" : ["/include", "/lib/pkgconfig",
+               "/share/pkfconfig", "/share/aclocal",
+               "/man", "/share/man", "/share/gtk-doc",
+               "/share/vala",
+               "*.la", "*.a"],
+  "modules" : [
+    {
+      "name" : "vala",
+      "config-opts" : [
+        "--disable-valadoc"
+      ],
+      "sources" : [
+        {
+          "type" : "git",
+          "url" : "https://gitlab.gnome.org/GNOME/vala.git",
+          "tag" : "0.40.12"
+        },
+        {
+          "type" : "patch",
+          "path" : "patches/Allow-disabling-valadoc.patch"
+        }
+      ]
+    },
+    {
+      "name" : "gtksourceview-4",
+      "sources" : [
+        {
+          "type" : "git",
+          "url" : "https://gitlab.gnome.org/GNOME/gtksourceview.git",
+          "tag" : "4.0.3"
+        }
+      ]
+    },
+    {
+      "name" : "amtk",
+      "builddir" : true,
+      "sources" : [
+        {
+          "type" : "git",
+          "url" : "https://gitlab.gnome.org/GNOME/amtk.git",
+          "tag" : "5.0.0"
+        }
+      ]
+    },
+    {
+      "name" : "uchardet",
+      "buildsystem" : "cmake",
+      "sources" : [
+        {
+          "type" : "git",
+          "url" : "https://github.com/BYVoid/uchardet.git",
+          "tag" : "v0.0.5"
+        }
+      ]
+    },
+    {
+      "name" : "tepl",
+      "sources" : [
+        {
+          "type" : "git",
+          "url" : "https://gitlab.gnome.org/GNOME/tepl.git",
+          "tag" : "4.2.0"
+        }
+      ]
+    },
+    {
+      "name" : "gspell",
+      "sources" : [
+        {
+          "type" : "git",
+          "url" : "https://gitlab.gnome.org/GNOME/gspell.git",
+          "tag" : "1.8.1"
+        }
+      ]
+    },
+    {
+      "name" : "libgee",
+      "config-opts" : [
+        "--enable-introspection=no",
+        "--disable-static"
+      ],
+      "sources" : [
+        {
+          "type" : "git",
+          "url" : "https://gitlab.gnome.org/GNOME/libgee.git",
+          "tag" : "0.20.1"
+        }
+      ]
+    },
+    {
+      "name" : "gnome-latex",
+      "sources" : [
+        {
+          "type" : "git",
+          "url" : "https://gitlab.gnome.org/coldtom/gnome-latex.git",
+          "tag" : "3.30.2-flatpak"
```

Comment:
It seems that for some reason the flatpak-builder doesn't like this tag.

```
flatpak-builder --download-only --no-shallow-clone --allow-missing-runtimes --state-dir=/srv/buildbot/sources /srv/buildbot/sources/.builddir org.gnome.gnome-latex.json
 in dir /srv/buildbot/worker/download-sources/build (timeout 3600 secs)
 watching logfiles {}
 argv: [b'flatpak-builder', b'--download-only', b'--no-shallow-clone', b'--allow-missing-runtimes', b'--state-dir=/srv/buildbot/sources', b'/srv/buildbot/sources/.builddir', b'org.gnome.gnome-latex.json']
 using PTY: True
Can't parse 'org.gnome.gnome-latex.json': <data>:117:34: Parse error: invalid string constant "3.30.2-flatpak", expected valid string constant
program finished with exit code 1
elapsedTime=0.036017
```

Moreover, you have to specify the `commit` anyway, so just replace this `tag` with the corresponding `commit`.
```
	          "commit" : "7c50c4369be32b2f78938670d460232b2912814e"
```

In long term, you should try to put vendor repo here and use `rename-icon` or apply patches if needed.
https://github.com/flathub/flathub/pull/853#issuecomment-461751810

See also:
http://docs.flatpak.org/en/latest/flatpak-builder-command-reference.html#idm45850113011824

Reply:
I wasn't aware of `rename-icon` and using a fork wasn't sitting well with me anyway, thanks!

I also specify the commit now, so hopefully things will build properly.